packages/plugin-backgroundcolor/src/BackgroundColor.ts

```diff
@@ -0,0 +1,164 @@
+import { JWPlugin, JWPluginConfig } from '../../core/src/JWPlugin';
+import { CommandParams } from '../../core/src/Dispatcher';
+import { VNode } from '../../core/src/VNodes/VNode';
+import { Loadables } from '../../core/src/JWEditor';
+import { Keymap } from '../../plugin-keymap/src/Keymap';
+import { Inline } from '../../plugin-inline/src/Inline';
+import { InlineNode } from '../../plugin-inline/src/InlineNode';
+import { getStyles, setStyles, setStyle, getStyle, removeStyle } from '../../utils/src/utils';
+import { VRange } from '../../core/src/VRange';
+import { Format } from '../../plugin-inline/src/Format';
+
+export interface ColorParams extends CommandParams {
+    color?: string; // css-valid color name
+}
+
+interface BackgroundColorConfig extends JWPluginConfig {
+    defaultBackgroundColor?: string;
+}
+
+const defaultConfiguration = {
+    defaultBackgroundColor: 'white',
+};
+
+export class BackgroundColor<T extends BackgroundColorConfig = BackgroundColorConfig> extends JWPlugin<T> {
+    static hasColor(node: VNode): boolean;
+    static hasColor(backgroundColor: string, node: VNode): boolean;
+    static hasColor(backgroundColor: string | VNode, node?: VNode): boolean {
+        if (backgroundColor instanceof VNode) {
+            node = backgroundColor;
+        }
+        const nodeBackgroundColor = getStyle(node, 'background-color');
+        if (!nodeBackgroundColor || backgroundColor instanceof VNode) {
+            return !!nodeBackgroundColor;
+        } else {
+            return nodeBackgroundColor === backgroundColor;
+        }
+    }
+    static dependencies = [Inline];
+    configuration = { ...defaultConfiguration, ...this.configuration };
+    commands = {
+        colorBackground: {
+            handler: this.colorBackground,
+        },
+        uncolorBackground: {
+            handler: this.uncolorBackground,
+        },
```

Comment:
I'm not 100% convinced by the command names. The `contenteditable` calls it `backColor` and DOM InputEvents call it `formatBackColor`. That does not convince me either. What about `setBackgroundColor` and `resetBackgroundColor` ? I think what bothers me is `uncolor`. Not sure... What do you think ?